test/unit_tests/util/RAIIWrappersTest.cpp

```diff
@@ -286,3 +286,36 @@ TEST(UtilsRAII, testGObjectSPtr) {
     }
     EXPECT_EQ(TestObjectCount, 0);
 }
+
+TEST(UtilsRAII, testGObjectSPtrReset) {
+    using TestPtr = xoj::util::GObjectSPtr<TestObject>;
+    EXPECT_EQ(TestObjectCount, 0);
+    {
+        TestPtr t;
+        t.reset(TEST_OBJECT(g_object_new(TEST_OBJECT_TYPE, nullptr)));
+        EXPECT_FALSE(g_object_is_floating(t.get()));
+        EXPECT_EQ(TestObjectCount, 1);
+    }
+    EXPECT_EQ(TestObjectCount, 0);
+    {
+        TestObject* t = TEST_OBJECT(g_object_new(TEST_OBJECT_TYPE, nullptr));
+        EXPECT_FALSE(g_object_is_floating(t));
+        g_object_force_floating(G_OBJECT(t));
+        EXPECT_TRUE(g_object_is_floating(t));
+        EXPECT_EQ(TestObjectCount, 1);
+        TestPtr t1;
+        t1.reset(t);  // Adopt a floating ref
+        EXPECT_EQ(t1.get(), t);
+        EXPECT_FALSE(g_object_is_floating(t1.get()));
+    }
+    EXPECT_EQ(TestObjectCount, 0);
+    {
+        TestObject* t = TEST_OBJECT(g_object_new(TEST_OBJECT_TYPE, nullptr));
+        EXPECT_EQ(TestObjectCount, 1);
+        TestPtr t1;
+        t1.reset(t, TestPtr::ref);
```

Comment:
Should assert a count of two here.

Reply:
I don't know how to access a GObject's ref-count. Is there a way I missed?